# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxHarfBuzzShaper.cpp
# Commit: 1c1f408004e1
# Full Hash: 1c1f408004e10207cb8e3479be397313f484bd3a
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1961387
# File Overlap Count: 1
# Description:
#   Bug 1961387 - Don't return a vertical-origin of zero for 'empty' truetype glyphs; fall back to computing the default vertical origin, so that if there's color-glyph data it will render in the right place. r=gfx-reviewers,lsalzman
#   
#   This fixes the bad positioning of Noto Color Emoji glyphs when using the
#   current (COLRv1) version of the font. GetGlyphVOrigin was incorrectly
#   returning zero because the 'glyf' data was empty; but the glyph will in
# ==============================================================================

diff -r c511ceb62edc -r 1c1f408004e1 gfx/thebes/gfxHarfBuzzShaper.cpp
--- a/gfx/thebes/gfxHarfBuzzShaper.cpp	Thu May 08 17:40:32 2025 +0000
+++ b/gfx/thebes/gfxHarfBuzzShaper.cpp	Thu May 08 17:42:10 2025 +0000
@@ -571,12 +571,10 @@
   if (mVmtxTable) {
     bool emptyGlyf;
     const Glyf* glyf = FindGlyf(aGlyph, &emptyGlyf);
-    if (glyf) {
-      if (emptyGlyf) {
-        *aY = 0;
-        return;
-      }
-
+    // If we didn't find any 'glyf' data, fall through to the default below;
+    // note that the glyph might still actually render (via SVG or COLR data),
+    // so we need to provide a reasonable origin.
+    if (glyf && !emptyGlyf) {
       const ::GlyphMetrics* metrics = reinterpret_cast<const ::GlyphMetrics*>(
           hb_blob_get_data(mVmtxTable, nullptr));
       int16_t lsb;
